n2kMappings.js

```diff
@@ -148,7 +184,196 @@ exports.mappings = {
     filter: function(n2k) {
       return n2k.fields['Engine Instance'] === 'Dual Engine Starboard';
     }
-  }],
+  }, {
+    source: 'Oil pressure',
+    node: 'propulsion.port.oilPressure',
+    filter: function(n2k) {
+      return n2k.fields['Engine Instance'] === 'Single Engine or Dual Engine Port';
+    }
+  }, {
+    source: 'Total Engine hours',
+    node: 'propulsion.port.runTime',
+    filter: function(n2k) {
+      return n2k.fields['Engine Instance'] === 'Single Engine or Dual Engine Port';
+    }
+  }, 
+   {
+    node: 'notifications.checkEngine',
```

Comment:
I've got something better, but still not what I would like:


```
   {
     node: function(n2k) { return engineAlarmPath(0) },
     value: function(n2k) { return engineAlarmsValue(0, n2k) }
   },
   {
     node: function(n2k) { return engineAlarmPath(1) },
     value: function(n2k) { return engineAlarmsValue(1, n2k) }
   },
...

var engineAlarms = [
  {
    path: 'notifications.checkEngine',
    message: 'Check Engine'
  },
  {
    path: 'notifications.engineOverTemperature',
    message: 'Engine Over Temperature'
  },
...

function engineAlarmPath(idx) {
   return engineAlarms[idx].path
}

function engineAlarmsValue(idx, n2k) {
  if (n2k.fields['Discrete Status 1'] != 0
      && ((n2k.fields['Discrete Status 1']) & (1<<idx)) ) {
    res = {
      message: engineAlarms[idx].message,
      source: 'sounder',
      state: 'alarm'
        };
    return res
  }
  return null;
}

```

What I really need is to be able to make the mapping a function that can return an array of values. I could probably hack the mapper to do it, but there's probably an elegant js way to do it?